torch/csrc/Stream.cpp

```diff
@@ -16,13 +16,14 @@ static PyObject* THPStream_pynew(
   int64_t stream_id = 0;
   int64_t device_index = 0;
   int64_t device_type = 0;
-  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays,clang-diagnostic-writable-strings)
-  static char* kwlist[] = {"stream_id", "device_index", "device_type", nullptr};
+  // NOLINTNEXTLINE(cppcoreguidelines-avoid-c-arrays,modernize-avoid-c-arrays)
+  constexpr const char* kwlist[] = {
+      "stream_id", "device_index", "device_type", nullptr};
   if (!PyArg_ParseTupleAndKeywords(
           args,
           kwargs,
           "|KKK",
-          kwlist,
+          reinterpret_cast<char**>(kwlist),
```

Comment:
Oh yeah, I see why these are writable strings now... https://docs.python.org/3/c-api/arg.html it doesn't have the appropiate const specifiers. They have to be made writable anyway to be fed into it. Still should be a const_cast not a reinterpret_cast. It's technically UB either way though so I think the original code is best as is.

Reply:
I checked the CPYTHON source code before.  The kwlist arrary is treated as a const one. There is no array reallocation inside the implementation so the memory is safe. We only need to handle these casts properly.

Reply:
const_cast should work